R/layer-sf.R

```diff
@@ -44,13 +44,9 @@ LayerSf <- ggproto("LayerSf", Layer,
 
     # automatically determine the name of the geometry column
     # and add the mapping if it doesn't exist
-    if ((isTRUE(self$inherit.aes) && is.null(self$computed_mapping$geometry) &&
-         is.null(plot$computed_mapping$geometry)) ||
-        (!isTRUE(self$inherit.aes) && is.null(self$computed_mapping$geometry))) {
-      if (is_sf(data)) {
-        geometry_col <- attr(data, "sf_column")
-        self$computed_mapping$geometry <- sym(geometry_col)
-      }
```

Comment:
If we define `A` as `self$inherit.aes` and `B` as `self$computed_mapping$geometry`, which the new class slots we'd be testing the following:

```r
if ((isTRUE(A) && is.null(B) && is.null(B)) || (!isTRUE(A) && is.null(B)))
```
which simplifies to:
```r
if ((isTRUE(A) && is.null(B)) || (!isTRUE(A) && is.null(B)))
```
which simplifies to:
```r
if ((isTRUE(A) || !isTRUE(A)) && is.null(B))
```
Which cancels out the OR operation because it will always return `TRUE`, thus we're left with:
```r
is.null(B)
```
I don't quite understand why the logic was hard to follow in the first place.

Reply:
oh, so I think actually it should have been `plot$mapping$geometry` and not `plot$computed_mapping$geometry`... but the global mapping has already conditionally been merged into `self$computed_mapping` so none of this even matters

you are right that it can all be simplified to `is.null(self$computed_mapping$geometry)`

Reply:
Yeah thanks for pointing out something was amiss in the logic here, this cleaned up really nicely ☺️